### PR TITLE
fix(report): account report uses ent currency

### LINKED ISSUE #924: Reports using Enterprise currency
This issue is to track ensuring all reports are provided in the enterprises set currency. 

**Required refactors**
- [x] Patient Invoice
- [ ] Cashflow (Requires controller refactor/ simplification for future maintenance) 
- [x] Patient financial activity
- [x] Customer Debts
- [x] Income vs. Expense 
- [x] Accounts Report

### PR DESCRIPTION
**Pull request by jniles**
This commit fixes account report to use the enterprise currency in
rendering totals.  It also migrates the account selection to a ui-select
for good measure.  Finally, the loading indicator on the form is now
  hooked up.

Partially fixes #924.

----
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
@sfount, this can be reviewed anytime.

**Comment by jniles** [comment]
@sfount, can I get a review?

File: server/controllers/finance/reports/reportAccounts/index.js [hunk 1]
```diff
@@ -67,54 +62,49 @@ function document(req, res, next) {
...
-function queryReportAccount(accountId, source){
+*/
+function queryReportAccount(accountId, source) {
   source = parseInt(source);
+  let sql;
 
-  if(source === 1){
-
-    let sql = `
+  if (source === 1) {
```
Comment (sfount) [inline server/controllers/finance/reports/reportAccounts/index.js:70]: There is no indication at all here as to what source is supposed to mean or refer to - it looks like it's taking values from either the posting journal, general ledger or both however this is not at all clear given the code. 

Should this be refactored here or do you think it is inappropriate for this pull request?

File: server/controllers/finance/reports/reportAccounts/index.js [hunk 2]
```diff
@@ -67,54 +62,49 @@ function document(req, res, next) {
...
       GROUP BY posting_journal.trans_id
-      ORDER BY posting_journal.trans_date ASC;`;
-
-    return db.exec(sql, [accountId]);
-
-  } else if(source === 3){
-    let sql = `
+      ORDER BY posting_journal.trans_date ASC;
+    `;
+  } else if (source === 3) {
```
Comment (sfount) [inline server/controllers/finance/reports/reportAccounts/index.js:88]: Is there not a MySQL view that combines the posting journal/ general ledger without manually doing that each time?

**Review by sfount — COMMENTED** [review]
Changes for making currency flexible LGTM. I have made a few comments on the server controller for this report. These comments may or may not be appropriate for this pull request.

File: server/controllers/finance/reports/reportAccounts/index.js (hunk 2 above)
Comment (jniles) [reply server/controllers/finance/reports/reportAccounts/index.js:88]: Yup!  It's called the `combined_ledger`.  Would you like me to do a refactor on this more than just the currency?

File: server/controllers/finance/reports/reportAccounts/index.js (hunk 1 above)
Comment (jniles) [reply server/controllers/finance/reports/reportAccounts/index.js:70]: In my view, I would do that as a separate PR.  I have no idea if these queries even produce correct data - a separate PR could do a full audit.

File: server/controllers/finance/reports/reportAccounts/index.js (hunk 2 above)
Comment (sfount) [reply server/controllers/finance/reports/reportAccounts/index.js:88]: Not in this pull request - as you mentioned this report could use a refactor/ audit. Could you make an issue to track this?

**Pull request merged**